src/pepr/operator/index.ts

```diff
@@ -98,6 +98,25 @@ When(UDSPackage)
     UDSConfig.isIdentityDeployed = false;
   });
 
+// Watch for optional Ambient mode and update config (temporary while ambient components are optional)
+When(a.DaemonSet)
+  .IsCreatedOrUpdated()
+  .InNamespace("istio-system")
+  .WithName("ztunnel")
+  .Watch(() => {
+    // todo: wait for keycloak and authservice to be running?
+    log.info("Istio Ambient deployed, operator configured to handle ambient packages.");
+    UDSConfig.isAmbientDeployed = true;
+  });
+When(a.DaemonSet)
+  .IsDeleted()
+  .InNamespace("istio-system")
+  .WithName("ztunnel")
+  .Watch(() => {
+    log.info("Istio Ambient deployed, operator will run all packages in sidecar mode.");
```

Comment:
Do we need to wait for Keycloak and Authservice here as well?

Reply:
ooh I see....

The comment in line 107 is the a bit misleading as it says:

```
// todo: wait for keycloak and authservice to be running?
```

So the question is how do we want to proceed here? Do we care only about ztunnel or do we want to ensure the Auth stack is also deployed when someone "flips the switch"?